onchain/src/social/deposit.cairo

```diff
@@ -1,20 +1,42 @@
+use core::fmt::Display;
+use core::to_byte_array::FormatAsByteArray;
 use starknet::{get_caller_address, get_contract_address, get_tx_info, ContractAddress};
 use super::request::{SocialRequest, SocialRequestImpl, SocialRequestTrait, Encode, Signature};
 
 pub type DepositId = felt252;
 
+// TODO add starknet_recipient as a Contract address
+// Find a way to format ContractAddress
+#[derive(Clone, Debug, Drop, Serde)]
+pub struct ClaimContent {
+    pub deposit_id: DepositId,
+    pub starknet_recipient: felt252
+// pub starknet_recipient: ContractAddress
+}
+
+
 impl DepositIdEncodeImpl of Encode<DepositId> {
     fn encode(self: @DepositId) -> @ByteArray {
         @format!("claim {}", self)
     }
 }
 
+// TODO: Find a way to format ContractAddress
+// Implement Display trait for Contract address
+impl ClaimContentEncodeImpl of Encode<ClaimContent> {
+    fn encode(self: @ClaimContent) -> @ByteArray {
+        // let felt_recipient:felt252=self.starknet_recipient.format_as_byte_array(16);
+        @format!("claim {} to {:?}", self.deposit_id, self.starknet_recipient)
```

Comment:
Why `{:?}`?

Reply:
Without the {:?}, the ContractAddress can't be format and display. 
We need maybe to do a Display Format trait for ContractAddress?